File: wp-content/prerf/analyze-har-metrics.py
#!/usr/bin/env python3
"""
Analyze HAR files for performance metrics
"""

import json
import glob
import os
from pathlib import Path

def analyze_har_file(file_path):
    """Analyze a single HAR file for metrics"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            har_data = json.load(f)
    except Exception as e:
        print(f"Error reading {file_path}: {e}")
        return None
    
    if 'log' not in har_data or 'entries' not in har_data['log']:
        return None
    
    metrics = {
        'total_requests': 0,
        '404_errors': 0,
        'css_files': 0,
        'js_files': 0,
        'recaptcha_loads': 0,
        'blocked_domains': {},
        'file_path': os.path.basename(file_path),
        '404_urls': []
    }
    
    blocked_domains = ['elementor', 'yith', 'revslider', 'revolution', 'slider']
    
    for entry in har_data['log']['entries']:
        metrics['total_requests'] += 1
        
        url = entry.get('request', {}).get('url', '')
        status = entry.get('response', {}).get('status', 0)
        mime_type = entry.get('response', {}).get('content', {}).get('mimeType', '')
        
        # Check 404 errors
        if status == 404:
            metrics['404_errors'] += 1
            metrics['404_urls'].append(url)
        
        # Count CSS files
        if 'css' in mime_type or '.css' in url:
            metrics['css_files'] += 1
        
        # Count JS files
        if 'javascript' in mime_type or '.js' in url:
            metrics['js_files'] += 1
        
        # Check for reCAPTCHA
        if 'recaptcha' in url.lower() or 'grecaptcha' in url.lower():
            metrics['recaptcha_loads'] += 1
        
        # Check for blocked domains that are still loading
        for domain in blocked_domains:
            if domain in url.lower():
                metrics['blocked_domains'][domain] = metrics['blocked_domains'].get(domain, 0) + 1
    
    return metrics

def main():
    har_dir = '/mnt/f/Obsidian/Dữ Liệu Của Tôi/Quản Lý Tài Chính/Quản Lý Kinh Doanh bán Tool/Vidieu.vn/public_html/wp-content/prerf/inputs/'
    har_files = glob.glob(os.path.join(har_dir, '*.har'))
    
    results = {}
    focus_pages = ['trang-cart.har', 'trang-checkout.har', 'trang-order-received.har']
    
    # Analyze each HAR file
    for har_file in har_files:
        basename = os.path.basename(har_file)
        metrics = analyze_har_file(har_file)
        if metrics:
            results[basename] = metrics
    
    # Output results
    print("=== HAR Files Analysis Results ===\n")
    
    # Summary table for Cart, Checkout, and Order-received pages
    print("### Focus: Cart, Checkout, and Order-received Pages\n")
    print("| Page | Total Requests | 404 Errors | CSS Files | JS Files | reCAPTCHA | Blocked Domains Still Loading |")
    print("|------|----------------|------------|-----------|----------|-----------|------------------------------|")
    
    for page in focus_pages:
        if page in results:
            m = results[page]
            blocked_domains_str = ''
            if m['blocked_domains']:
                blocked_items = [f"{domain}({count})" for domain, count in m['blocked_domains'].items()]
                blocked_domains_str = ', '.join(blocked_items)
            
            page_name = page.replace('.har', '')
            print(f"| {page_name:<20} | {m['total_requests']:>14} | {m['404_errors']:>10} | {m['css_files']:>9} | {m['js_files']:>8} | {m['recaptcha_loads']:>9} | {blocked_domains_str or 'None':<28} |")
    
    print("\n### All Pages Summary\n")
    print("| Page | Total Requests | 404 Errors | CSS Files | JS Files | reCAPTCHA | Blocked Domains Still Loading |")
    print("|------|----------------|------------|-----------|----------|-----------|------------------------------|")
    
    for page, m in results.items():
        blocked_domains_str = ''
        if m['blocked_domains']:
            blocked_items = [f"{domain}({count})" for domain, count in m['blocked_domains'].items()]
            blocked_domains_str = ', '.join(blocked_items)
        
        page_name = page.replace('.har', '')
        print(f"| {page_name:<20} | {m['total_requests']:>14} | {m['404_errors']:>10} | {m['css_files']:>9} | {m['js_files']:>8} | {m['recaptcha_loads']:>9} | {blocked_domains_str or 'None':<28} |")
    
    # Detailed 404 errors for focus pages
    print("\n### 404 Errors Details for Focus Pages\n")
    for page in focus_pages:
        if page in results and results[page]['404_urls']:
            print(f"**{page.replace('.har', '')}:**")
            for url in results[page]['404_urls']:
                print(f"- {url}")
            print()

if __name__ == "__main__":
    main()

File: wp-content/prerf/analyze-har-metrics.php
<?php
/**
 * Analyze HAR files for performance metrics
 */

function analyzeHarFile($filePath) {
    $harContent = file_get_contents($filePath);
    $harData = json_decode($harContent, true);
    
    if (!$harData || !isset($harData['log']['entries'])) {
        return null;
    }
    
    $metrics = [
        'total_requests' => 0,
        '404_errors' => 0,
        'css_files' => 0,
        'js_files' => 0,
        'recaptcha_loads' => 0,
        'blocked_domains' => [],
        'file_path' => basename($filePath)
    ];
    
    $blockedDomains = [
        'elementor' => 0,
        'yith' => 0,
        'revslider' => 0,
        'revolution' => 0,
        'slider' => 0
    ];
    
    foreach ($harData['log']['entries'] as $entry) {
        $metrics['total_requests']++;
        
        $url = $entry['request']['url'] ?? '';
        $status = $entry['response']['status'] ?? 0;
        $mimeType = $entry['response']['content']['mimeType'] ?? '';
        
        // Check 404 errors
        if ($status == 404) {
            $metrics['404_errors']++;
        }
        
        // Count CSS files
        if (strpos($mimeType, 'css') !== false || strpos($url, '.css') !== false) {
            $metrics['css_files']++;
        }
        
        // Count JS files
        if (strpos($mimeType, 'javascript') !== false || strpos($url, '.js') !== false) {
            $metrics['js_files']++;
        }
        
        // Check for reCAPTCHA
        if (stripos($url, 'recaptcha') !== false || stripos($url, 'grecaptcha') !== false) {
            $metrics['recaptcha_loads']++;
        }
        
        // Check for blocked domains that are still loading
        foreach ($blockedDomains as $domain => $count) {
            if (stripos($url, $domain) !== false) {
                $blockedDomains[$domain]++;
            }
        }
    }
    
    // Filter out domains with 0 hits
    $metrics['blocked_domains'] = array_filter($blockedDomains);
    
    return $metrics;
}

// Analyze all HAR files
$harDir = '/mnt/f/Obsidian/Dữ Liệu Của Tôi/Quản Lý Tài Chính/Quản Lý Kinh Doanh bán Tool/Vidieu.vn/public_html/wp-content/prerf/inputs/';
$harFiles = glob($harDir . '*.har');

$results = [];
$focusPages = ['trang-cart.har', 'trang-checkout.har', 'trang-order-received.har'];

// Analyze each HAR file
foreach ($harFiles as $harFile) {
    $basename = basename($harFile);
    $metrics = analyzeHarFile($harFile);
    
    if ($metrics) {
        $results[$basename] = $metrics;
    }
}

// Output results
echo "=== HAR Files Analysis Results ===\n\n";

// Summary table for Cart, Checkout, and Order-received pages
echo "### Focus: Cart, Checkout, and Order-received Pages\n\n";
echo "| Page | Total Requests | 404 Errors | CSS Files | JS Files | reCAPTCHA | Blocked Domains Still Loading |\n";
echo "|------|----------------|------------|-----------|----------|-----------|------------------------------|\n";

foreach ($focusPages as $page) {
    if (isset($results[$page])) {
        $m = $results[$page];
        $blockedDomainsStr = '';
        if (!empty($m['blocked_domains'])) {
            $blockedDomainsStr = [];
            foreach ($m['blocked_domains'] as $domain => $count) {
                $blockedDomainsStr[] = "$domain($count)";
            }
            $blockedDomainsStr = implode(', ', $blockedDomainsStr);
        }
        
        echo sprintf("| %-20s | %-14d | %-10d | %-9d | %-8d | %-9d | %-28s |\n",
            str_replace('.har', '', $m['file_path']),
            $m['total_requests'],
            $m['404_errors'],
            $m['css_files'],
            $m['js_files'],
            $m['recaptcha_loads'],
            $blockedDomainsStr ?: 'None'
        );
    }
}

echo "\n### All Pages Summary\n\n";
echo "| Page | Total Requests | 404 Errors | CSS Files | JS Files | reCAPTCHA | Blocked Domains Still Loading |\n";
echo "|------|----------------|------------|-----------|----------|-----------|------------------------------|\n";

foreach ($results as $page => $m) {
    $blockedDomainsStr = '';
    if (!empty($m['blocked_domains'])) {
        $blockedDomainsStr = [];
        foreach ($m['blocked_domains'] as $domain => $count) {
            $blockedDomainsStr[] = "$domain($count)";
        }
        $blockedDomainsStr = implode(', ', $blockedDomainsStr);
    }
    
    echo sprintf("| %-20s | %-14d | %-10d | %-9d | %-8d | %-9d | %-28s |\n",
        str_replace('.har', '', $m['file_path']),
        $m['total_requests'],
        $m['404_errors'],
        $m['css_files'],
        $m['js_files'],
        $m['recaptcha_loads'],
        $blockedDomainsStr ?: 'None'
    );
}

// Detailed 404 errors for focus pages
echo "\n### 404 Errors Details for Focus Pages\n\n";
foreach ($focusPages as $page) {
    $harFile = $harDir . $page;
    if (file_exists($harFile)) {
        $harContent = file_get_contents($harFile);
        $harData = json_decode($harContent, true);
        
        $errors404 = [];
        foreach ($harData['log']['entries'] as $entry) {
            if (($entry['response']['status'] ?? 0) == 404) {
                $errors404[] = $entry['request']['url'];
            }
        }
        
        if (!empty($errors404)) {
            echo "**" . str_replace('.har', '', $page) . ":**\n";
            foreach ($errors404 as $url) {
                echo "- $url\n";
            }
            echo "\n";
        }
    }
}